File: webapp/app/models/domain.ts
export class Domain {
  id: number;
  name: string;
  format: string;
  description: string;
  existential: string;
  created: Date;
  updated: Date;

  constructor(
    id?,
    name?,
    format?,
    description?,
    existential?,
    created?,
    updated?,
  ) {
    this.id = id ? id : 1;
    this.name = name ? name : '';
    this.format = format ? format : '';
    this.description = description ? description : '';
    this.existential = existential ? existential : '';
    this.created = created ? created : new Date();
    this.updated = updated ? updated : new Date();
  }
}

File: src/main/java/org/enlightenseries/DomainDictionary/domain/model/domain/Domain.java
package org.enlightenseries.DomainDictionary.domain.model.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Domain {
  private Long id;
  private String name;
  private String format;
  private String description;
  private String existential;
  private Date created;
  private Date updated;
}
